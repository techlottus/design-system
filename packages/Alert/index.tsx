import * as React from 'react';
import classNames from 'classnames';
import { IProps } from '../common/props';
import style from './alert.module.pcss';
import Icon from '../Icon';

interface IAlertProps extends IProps {
  children?: React.ReactNode;
  closable?: boolean;
  show?: boolean;
  onClose?: () => void;
  bordered?: boolean;
  type: 'default'| 'error' | 'success' | 'warning' | 'info';
}

interface IAlertState {
  closed: boolean;
}

const messageColorTypes = {
  error: classNames('bg-primary-500', 'text-white'),
  warning: classNames('bg-warning'),
  success: classNames('bg-success'),
  default: classNames('bg-surface-100'),
};

class Alert extends React.PureComponent<IAlertProps, IAlertState> {
  static defaultProps: IAlertProps = {
    bordered: true,
    type: 'default',
    closable: false,
  };

  state: IAlertState = {
    closed: false,
  };

  onCloseHandler = () => {
    this.setState({
      closed: !this.state.closed,
    });
  }

  render() {
    const { className, type, children, closable } = this.props;
    const classes = classNames(
      style.alert,
      className,
      messageColorTypes[type],
      {
        [style.alertWithIcon]: closable,
      },
    );

    return (
      !this.state.closed ? <div className={classes}>
        {children}
        { closable ? <button onClick={this.onCloseHandler}>
          <Icon className={style.alertIcon} size={20} icon='icon-close' />
        </button> : null }
      </div> : null
    );
  }
}

export default Alert;
