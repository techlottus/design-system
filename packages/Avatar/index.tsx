import * as React from 'react';
import { IProps } from '../common/props';
import classNames from 'classnames';

const Sizes = {
  'xs': classNames('w-4 h-4'),
  'sm': classNames('w-6 h-6'),
  'md': classNames('w-8 h-8'),
  'lg': classNames('w-10 h-10'),
  'xl': classNames('w-12 h-12'),
  '2xl': classNames('w-14 h-14'),
};

type Size = typeof Sizes[keyof typeof Sizes];

export interface IAvatar extends IProps {
  children?: React.ReactElement;
  size?: Size;
}

const Avatar: React.SFC<IAvatar> = ({ size, children, className }: IAvatar) => {
  const classes = classNames('relative', className, Sizes[size]);

  return (
    <figure className={classes}>
      {React.cloneElement(
        children,
        {
          className:'rounded',
        },
        null
      )}
    </figure>
  );
};

Avatar.defaultProps = {
  size: 'md',
};

export default Avatar;
