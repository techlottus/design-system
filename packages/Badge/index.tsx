import * as React from 'react';
import { IProps } from '../common/props';
import classNames from 'classnames';

export interface IBadge extends IProps {
  content?: string;
  roundedFull?: boolean;
}

const Badge: React.SFC<IBadge> = ({ className, content, roundedFull }: IBadge) => {
  const fullBadgeClasses = classNames('flex justify-center items-center rounded-full h-9 w-9');
  const classes = classNames(
    'inline-block px-2 py-1 rounded bg-primary-500 text-white text-xs font-rubik',
    className,
    { [fullBadgeClasses]: roundedFull },
  );
  return (
    <span
      className={classes}>
      {content}
    </span>
  );
};

Badge.defaultProps = {
  content: 'Nuevo curso',
  roundedFull: false,
};

export default Badge;
