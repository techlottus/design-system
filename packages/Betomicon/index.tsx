import * as React from 'react';

// Icon Props
interface BetomiconProps {
  icon: string;
  size?: string;
  className?: string;
}

const Icon: React.SFC<BetomiconProps> = ({ size, icon, className }: BetomiconProps) => {
  return(
    <div>
      <i style={{fontSize: size}}>{icon}</i>
    </div>
  );
};

Icon.defaultProps = {
  icon: 'ðŸ˜‡',
  size: '12px',
};

export default Icon;
