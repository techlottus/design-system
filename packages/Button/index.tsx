import * as React from 'react';
import Icon from '../Icon';

// Button Props
const sizes = {
  xs: '',
  sm: '',
  md: 'px-8 py-3 text-base',
  lg: 'px-10 py-5 text-xl',
};

const buttonVariant = {
  primary: `
    bg-primary-400
    hover:bg-primary-700
    inline-flex
    items-center
    rounded
    text-white
    font-bold
    font-montserrat
  `,
  secondary: `
    bg-surface-500
    hover:bg-surface-800
    inline-flex
    items-center
    rounded
    text-white
    font-bold
    font-montserrat
  `,
  terciary: `
    bg-transparent
    hover:bg-surface-400
    hover:text-white
    border-2
    border-surface-400
    rounded
    inline-flex
    items-center
    font-bold
    font-montserrat
  `,
  disabled: `
    bg-surface-100
    rounded
    text-surface-300
    font-bold
    font-montserrat
    cursor-not-allowed
  `,
};

interface ButtonProps {
  size: 'xs'|'sm'|'md'|'lg';
  variant: 'primary'|'secondary'|'terciary'|'disabled';
  children: any;
  className?: string;
}

const Button: React.SFC<ButtonProps> = ({ size, variant, children, className }: ButtonProps) => (
  <button className={`${className} ${sizes[size]} ${buttonVariant[variant]}`}>{children}</button>
);

Button.defaultProps = {
  size: 'md',
  variant: 'primary',
  className: '',
};

export default Button;
