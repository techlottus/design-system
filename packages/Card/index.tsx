import * as React from 'react';
import classNames from '../common/classnames';
import { variantIntensity, variantName } from '../common/constants';
import CardHeader from './CardHeader';
import CardContent from './CardContent';
import CardFooter from './CardFooter';
import styles from './card.module.pcss';

export interface ICardProps {
  children?: React.ReactNode;
  bordered?: boolean;
  variant?: string;
  variantAs?: string;
}

class Card extends React.Component<ICardProps> {
  public static Header = CardHeader;
  public static Content = CardContent;
  public static Footer = CardFooter;
  static defaultProps: { bordered: boolean; variant: string; variantAs: string; };

  public render() {
    const { children, bordered, variant, variantAs } = this.props;
    const classes = classNames(
      styles.cardBordered,
      `border-${variantName[variant]}-${variantIntensity[variantAs]}`,
    );

    return(
      <div className={styles.card}>
        {bordered && <div className={classes} />}
        {children}
      </div>
    );
  }
}

Card.defaultProps = {
  bordered: false,
  variant: 'surface',
  variantAs: '500',
};

export default Card;
