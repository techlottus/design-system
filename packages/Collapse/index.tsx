import * as React from 'react';
import classNames from 'classnames';
import { IProps } from '../common/props';
import style from './collapse.module.pcss';
import Icon from '../Icon/index';

export interface ICollapse extends IProps {
  children?: React.ReactNode;
  title?: string;
  iconOpened?: string;
  iconClosed?: string;
  isOpen?: boolean;
}

export interface ICollapseState {
  isOpen?: boolean;
}

class Collapse extends React.PureComponent<ICollapse, ICollapseState> {
  static defaultProps: ICollapse = {
    iconOpened: 'icon-minus',
    iconClosed: 'icon-plus',
    isOpen: false,
  };

  public state = {
    isOpen: false,
  };

  // Handler function for toggle
  onToggle = () => {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  }

  // Handler element for create toggable icon
  handlerElement = () => (
    <button className='flex' onClick={this.onToggle}>
      { this.state.isOpen
        ? <Icon className='text-secondary-500' size={24} icon='icon-minus' />
        : <Icon className='text-secondary-500' size={24} icon='icon-plus' /> }
    </button>
  )

  componentDidMount() {
    if (this.props.isOpen) {
      this.setState({
        isOpen: this.props.isOpen,
      });
    }
  }

  public render() {
    const { children, title, className } = this.props;
    const classes = classNames(style.collapse, className);

    return (
      <div className={classes}>
        <div className={style.collapseHeading}>
          <div className={style.collapseTitle}>{title}</div>
          <div className={style.collapseIcon}>
            {this.handlerElement()}
          </div>
        </div>
        {this.state.isOpen && <div className={style.collapseContent}>
          {children}
        </div>}
      </div>
    );
  }
}

export default Collapse;
