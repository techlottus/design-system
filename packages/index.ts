import './tailwind.css';

export { default as Button } from './Button';
export { default as Card } from './Card';
export { default as FormGroup } from './FormGroup';
export { default as GlideContainer } from './Glide';
export { default as Icon } from './Icon';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as ProgressBar } from './ProgressBar';
export { default as Nav } from './Nav';
export { default as Navbar } from './Navbar';
export { default as Container } from './Container';
export { default as Avatar } from './Avatar';
export { default as Collapse } from './Collapse';
