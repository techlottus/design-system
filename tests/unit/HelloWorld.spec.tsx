import * as React from "react";
import { shallow } from 'enzyme';
import {HelloWorld} from '../packages';

test('Create HelloWorld component', () => {
  const component = shallow(
    <HelloWorld message="Hello everybody!"  />,
  );
  expect(component).toMatchSnapshot();
});