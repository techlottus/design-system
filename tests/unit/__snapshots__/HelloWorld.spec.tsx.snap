// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.tsx should match the snapshot 1`] = `
<h1>
  Hello everybody!
</h1>
`;
